guides/release/testing/testing-application.md

```diff
@@ -44,9 +44,9 @@ module('Acceptance | posts', function(hooks) {
 
   test('should add new post', async function(assert) {
     await visit('/posts/new');
-    await fillIn('input.title', 'My new post');
-    await click('button.submit');
-    assert.equal(document.querySelector('ul.posts li').textContent, 'My new post');
+    await fillIn('[data-test-post-input-title]', 'My new post');
+    await click('[data-test-post-save]');
+    assert.dom('[data-test-post-title]').hasText('My new post');
```

Comment:
Good use of QUnit DOM. 👍🏼

In general, I like to recommend describing the assertion so that the test is readable to any developer, regardless of their knowledge of the component/route. Here's another optional suggestion:

```suggestion
    assert.dom('[data-test-post-title]')
      .hasText('My new post', 'The user sees the correct title.');
```